src/rocksdb/src/lib.rs

```diff
@@ -663,6 +663,16 @@ fn rocksdb_core_loop<K, V, M, O, IM>(
                 for _ in buf_size..batch_size {
                     encoded_batch_buffers.push(Some(Vec::new()));
                 }
+                // shrinking the buffers in case the scratch buffer's capacity is more than
+                // double the batch size
+                let half_capacity = encoded_batch_buffers.capacity() / 2;
+                if half_capacity > batch_size {
+                    encoded_batch_buffers.truncate(half_capacity);
```

Comment:
we `.clear()` before we use it, so we shouldn't need to truncate right?

Reply:
We clear the vec of bytes within this vec https://github.com/MaterializeInc/materialize/pull/21232/files#diff-8e949ab69256efd04acb76b94b8ee2668d4a14501e7cc3a6d0325572a39b268fR688, but not these ones directly. 

We do drain `encoded_batch` later though but don't remove anything from `encoded_batch_buffers`. So can remove `encoded_batch.truncate` but keep the other one.